File: my-folder/0020-valid-parentheses/solution.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const stack = [];
    for (let c of s) {
        if (c === '(' || c === '{' || c === '[') {
            stack.push(c);
        } else {
            const top = stack.pop()
            if (top === '(' && c === ')' || 
                top === '{' && c === '}' || 
                top === '[' && c === ']') {
            } else {
                return false;
            }
        }
    }
    return stack.length === 0;
};

File: my-folder/0020-valid-parentheses/solution.rs
impl Solution {
    pub fn is_valid(s: String) -> bool {
        let mut stack = vec![];
        for c in s.chars() {
            if c == '(' || c == '{' || c == '[' {
                stack.push(c);
            } else {
                if let Some(top) = stack.pop() {
                    if top == '(' && c == ')'||
                       top == '{' && c == '}'||
                       top == '[' && c == ']' {                       
                       } else {
                           return false;
                       }
            } else {
                return false;
            }
            }
        }
        stack.len() == 0
    }
}
